sdk/rust/oak_derive/src/lib.rs

```diff
@@ -59,24 +59,20 @@ pub fn derive_oak_exports(input: TokenStream) -> TokenStream {
 
     let expanded = quote! {
         #[no_mangle]
-        pub extern "C" fn oak_main(handle: u64) -> i32 {
+        pub extern "C" fn oak_main(handle: u64) {
             // A panic in the Rust module code cannot safely pass through the FFI
-            // boundary, so catch any panics here and translate to an error return.
+            // boundary, so catch any panics here and drop them.
             // https://doc.rust-lang.org/nomicon/ffi.html#ffi-and-panics
-            std::panic::catch_unwind(||{
+            let _ = std::panic::catch_unwind(||{
                 oak::set_panic_hook();
                 inner_main(handle)
-            })
-            .unwrap_or(Err(oak::OakStatus::ERR_INTERNAL))
-            .err()
-            .unwrap_or(oak::OakStatus::OK)
-            .value()
+            });
         }
         // Internal version of the main entrypoint, to allow testing without any
         // panic interception.
-        pub fn inner_main(handle: u64) -> Result<(), oak::OakStatus> {
+        pub fn inner_main(handle: u64) {
             let mut node = <#name>::new();
-            oak::grpc::event_loop(node, oak::ReadHandle{ handle: oak::Handle::from_raw(handle) })
+            let _ = oak::grpc::event_loop(node, oak::ReadHandle{ handle: oak::Handle::from_raw(handle) });
```

Comment:
This should also `unwrap` / `expect` the `Result` here so it gets logged, or it will be lost?

Reply:
As with the chat example, the `grpc::event_loop` will only ever return an `Err`, so `expect`ing anything means that Node termination always implies a `panic`.   However, I'm not keen on adding a log statement here because it induces a dependency on the `log` crate for anything that uses `oak_derive`.

Reply:
Couldn't the node terminate gracefully with `Ok` if it wanted to? e.g. in response to a specific gRPC message, or after some time, or for whatever other reason.

But even in the `event_loop`, we do sometimes propagate errors up, e.g.

https://github.com/project-oak/oak/blob/a0ab735c61fa69474b53878e8b9bf406c00a537f/sdk/rust/oak/src/grpc/mod.rs#L174

IIUC, after this commit, this error would just be ignored and never logged. Or am I missing something?

Reply:
`grpc::event_loop` does propagate errors up, but that's the only way that loop terminates (atm) &ndash; there's no path to emit `Ok`.  So any Node that uses `oak_derive` will always terminate its inner entrypoint with an `Err`, which would always trigger a `panic` if the `expect` is there, which will (I think) always generate a Wasm trap &ndash; even for orderly shutdown.

Reply:
(From discussion) We already transiently depend on the `log` crate anyway (`oak_derive` invokes `oak::set_panic_hook` which calls `log::error!`), so I've added a log here too.